### PR TITLE
[refactor] Move log2int into arithmatic.h

### PR DESCRIPTION
**Pull request by k-ye**
Related issue = N/A
WANT_LGTM=any

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

### CODE REVIEW

**Comment by k-ye** [comment]
I actually have a question. Can `runtime.cpp` include a header whose implementation is in `.cpp`..?

File: taichi/math/arithmetic.h [hunk 1]
```diff
@@ -14,4 +14,19 @@ T iroundup(T a, T b) {
   }
 }
 
+template <typename T>
+uint32_t log2int(T value) {
+  if constexpr (!std::is_integral_v<T>) {
+    static_assert(always_false_v<T>, "Must be integral type");
+  }
```
Comment (archibate) [inline taichi/math/arithmetic.h:21]: Why not just
```suggestion
  static_assert(std::is_integral_v<T>, "Must be integral type");
```
? Sorry, I'm not a C++ expert..

**Comment by archibate** [comment]
> I actually have a question. Can `runtime.cpp` include a header whose implementation is in `.cpp`..?

I don't think so. The `runtime.bc` will only contain codes from `runtime.cpp`, right? So we should make those impl inline in `.h`.

**Comment by k-ye** [comment]
> I don't think so. The runtime.bc will only contain codes from runtime.cpp, right? So we should make those impl inline in .h.

I also feel like so, which is why I made this a function template. But i guess it's easy to test out.

**Comment by archibate** [comment]
You may merge with master again to fix CI :)

**Pull request merged**